### ISSUE TITLE
gulp print errors

### ISSUE THREAD

**Issue by maimake** [ericmdantas/generator-ng-fullstack#197]
I create an app as following 👍 

? What is the name of your app? fullstack
? What is your username on Github? some-username-here
? What stack do you want? Full, client or server? fullstack
? What do you want in server side? node
? What do you want in client side? ng2
? What transpiler do you want to use in server side? typescript
? Do you want a secure app (http/2)? No
? Do you want to use a different static server? Such as nginx, apache, IIS? No

And then run `npm run dev` or just run `gulp`:

npm run dev

> fullstack@0.0.1 dev /Users/mai/Develop/learn/web/angular2/fullstack
> tsc && concurrently "npm start" "gulp"

node_modules/@angular/core/src/application_ref.d.ts(79,88): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/application_ref.d.ts(137,42): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/application_ref.d.ts(196,33): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/change_detection/differs/default_keyvalue_differ.d.ts(24,15): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/change_detection/differs/default_keyvalue_differ.d.ts(26,16): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/di/reflective_provider.d.ts(115,123): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/di/reflective_provider.d.ts(115,165): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/async.d.ts(34,33): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/async.d.ts(35,45): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/collection.d.ts(1,25): error TS2304: Cannot find name 'MapConstructor'.
node_modules/@angular/core/src/facade/collection.d.ts(2,25): error TS2304: Cannot find name 'SetConstructor'.
node_modules/@angular/core/src/facade/collection.d.ts(4,27): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/collection.d.ts(4,39): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/collection.d.ts(7,9): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/collection.d.ts(8,30): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/collection.d.ts(11,43): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/collection.d.ts(12,27): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/collection.d.ts(14,23): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/collection.d.ts(15,25): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/collection.d.ts(100,41): error TS2304: Cannot find name 'Set'.
node_modules/@angular/core/src/facade/collection.d.ts(101,22): error TS2304: Cannot find name 'Set'.
node_modules/@angular/core/src/facade/collection.d.ts(102,25): error TS2304: Cannot find name 'Set'.
node_modules/@angular/core/src/facade/lang.d.ts(11,17): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/lang.d.ts(12,17): error TS2304: Cannot find name 'Set'.
node_modules/@angular/core/src/facade/lang.d.ts(68,59): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/promise.d.ts(9,14): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/promise.d.ts(15,32): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/promise.d.ts(16,38): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/promise.d.ts(17,35): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/promise.d.ts(17,93): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/promise.d.ts(18,34): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/promise.d.ts(18,50): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/promise.d.ts(19,32): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/promise.d.ts(19,149): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/promise.d.ts(20,43): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/linker/compiler.d.ts(13,59): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/linker/component_resolver.d.ts(9,58): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/linker/component_resolver.d.ts(13,49): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/linker/dynamic_component_loader.d.ts(61,148): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/linker/dynamic_component_loader.d.ts(102,144): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/linker/dynamic_component_loader.d.ts(107,139): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/linker/dynamic_component_loader.d.ts(108,135): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/linker/systemjs_component_resolver.d.ts(18,53): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/linker/systemjs_component_resolver.d.ts(26,53): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/testing/component_fixture.d.ts(77,19): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/testing/test_component_builder.d.ts(74,57): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/forms/src/facade/async.d.ts(34,33): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/forms/src/facade/async.d.ts(35,45): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/forms/src/facade/promise.d.ts(9,14): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/forms/src/facade/promise.d.ts(15,32): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/forms/src/facade/promise.d.ts(16,38): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/forms/src/facade/promise.d.ts(17,35): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/forms/src/facade/promise.d.ts(17,93): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/forms/src/facade/promise.d.ts(18,34): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/forms/src/facade/promise.d.ts(18,50): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/forms/src/facade/promise.d.ts(19,32): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/forms/src/facade/promise.d.ts(19,149): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/forms/src/facade/promise.d.ts(20,43): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/http/src/headers.d.ts(45,59): error TS2304: Cannot find name 'Map'.
node_modules/@angular/http/src/url_search_params.d.ts(46,16): error TS2304: Cannot find name 'Map'.
node_modules/@angular/platform-browser-dynamic/index.d.ts(81,90): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/platform-browser-dynamic/index.d.ts(85,99): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/platform-browser-dynamic/index.d.ts(89,99): error TS2304: Cannot find name 'Promise'.
node_modules/rxjs/Observable.d.ts(10,66): error TS2304: Cannot find name 'Promise'.
node_modules/rxjs/Observable.d.ts(66,60): error TS2304: Cannot find name 'Promise'.
node_modules/rxjs/Observable.d.ts(66,70): error TS2304: Cannot find name 'Promise'.
server/api/todo/controller/todo-controller.ts(1,26): error TS2307: Cannot find module 'express'.
server/api/todo/dao/todo-dao.ts(1,27): error TS2307: Cannot find module 'mongoose'.
server/api/todo/dao/todo-dao.ts(2,26): error TS2307: Cannot find module 'bluebird'.
server/api/todo/dao/todo-dao.ts(3,20): error TS2307: Cannot find module 'lodash'.
server/api/todo/model/todo-model.ts(1,27): error TS2307: Cannot find module 'mongoose'.
server/api/todo/routes/todo-routes.ts(3,26): error TS2307: Cannot find module 'express'.
server/commons/static/index.ts(3,26): error TS2307: Cannot find module 'express'.
server/commons/static/index.ts(4,21): error TS2307: Cannot find module 'fs'.
server/commons/static/index.ts(8,19): error TS2304: Cannot find name 'process'.
server/config/db.conf.ts(3,27): error TS2307: Cannot find module 'mongoose'.
server/config/db.conf.ts(4,15): error TS2304: Cannot find name 'require'.
server/config/db.conf.ts(8,20): error TS2304: Cannot find name 'process'.
server/config/db.conf.ts(8,61): error TS2304: Cannot find name 'process'.
server/config/routes.conf.ts(3,26): error TS2307: Cannot find module 'express'.
server/config/routes.conf.ts(4,25): error TS2307: Cannot find module 'morgan'.
server/config/routes.conf.ts(5,29): error TS2307: Cannot find module 'body-parser'.
server/config/routes.conf.ts(6,25): error TS2307: Cannot find module 'helmet'.
server/config/routes.conf.ts(10,21): error TS2304: Cannot find name 'process'.
server/config/routes.conf.ts(12,29): error TS2304: Cannot find name 'process'.
server/routes/index.ts(1,26): error TS2307: Cannot find module 'express'.
server/server.ts(1,1): error TS6053: File 'typings/index.d.ts' not found.
server/server.ts(5,22): error TS2304: Cannot find name 'process'.
server/server.ts(6,5): error TS2304: Cannot find name 'require'.
server/server.ts(8,12): error TS2304: Cannot find name 'process'.
server/server.ts(10,26): error TS2307: Cannot find module 'express'.
server/server.ts(11,21): error TS2307: Cannot find module 'os'.
server/server.ts(12,23): error TS2307: Cannot find module 'http'.
server/server.ts(26,34): error TS2304: Cannot find name 'process'.
tests/client/todo/components/todo-cmp_test.ts(1,1): error TS6053: File 'typings/index.d.ts' not found.
tests/client/todo/components/todo-cmp_test.ts(62,9): error TS2304: Cannot find name 'spyOn'.
tests/client/todo/services/todo-service_test.ts(1,1): error TS6053: File 'typings/index.d.ts' not found.

npm ERR! Darwin 15.0.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "dev"
npm ERR! node v6.2.2
npm ERR! npm  v3.9.5
npm ERR! code ELIFECYCLE
npm ERR! fullstack@0.0.1 dev: `tsc && concurrently "npm start" "gulp"`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the fullstack@0.0.1 dev script 'tsc && concurrently "npm start" "gulp"'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the fullstack package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     tsc && concurrently "npm start" "gulp"
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs fullstack
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls fullstack
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/mai/Develop/learn/web/angular2/fullstack/npm-debug.log

**Comment by ericmdantas** [comment]
Hello, @maimake.

Make sure you run `typings install` before running `npm run dev` - there's a topic in the [wiki](https://github.com/ericmdantas/generator-ng-fullstack/wiki/Troubleshooting) about it.

**Comment by maimake** [comment]
You saved my time, thank you

**Comment by ericmdantas** [comment]
No problem :smile:

**Comment by maimake** [comment]
Could you add a post script in package.json if it's a typescript project ?
 
```
{
  "scripts": {
  	...
    "postinstall": "typings install",
    ...
  }
}
```

check [angular/quickstart](https://github.com/angular/quickstart/blob/master/package.json)

**Comment by ericmdantas** [comment]
It's added in the 1.9 version, but since I still have some stuff to do there, I'll pull that into master too. Thanks for the suggestion.

**Comment by ericmdantas** [comment]
Released 1.8.16 :smile: 

```json
// other
"preinstall": "typings install",
"postinstall": "tsc",
// other
```
